tilesets/scripts/cli.py

```diff
@@ -264,94 +276,75 @@ def update_recipe(tileset, recipe, token=None):
-def validate_source(source_path):
+@cligj.features_in_arg
+def validate_source(features):
     """Validate your source file.
     $ tilesets validate-source <path/to/your/src/file>
     """
-    line_count = 1
-    with open(source_path, "r") as inf:
-        click.echo("Validating {0} ...".format(source_path))
-        feature = None
-        try:
-            for feature in JSONSeqDecoder().decode(inf):
-                utils.validate_geojson(feature)
-                line_count += 1
-        except JSONDecodeError:
-            click.echo(
-                "Error: Invalid JSON on line {} \n Invalid Content: {} \n".format(
-                    line_count, feature
-                )
-            )
-            sys.exit(1)
-        except jsonschema.exceptions.ValidationError:
-            click.echo(
-                "Error: Invalid geojson found on line {} \n Invalid Feature: {} \n Note - Geojson must be line delimited.".format(
-                    line_count, feature
-                )
-            )
-            sys.exit(1)
+    click.echo(f"Validating features", err=True)
+
+    for feature in features:
+        utils.validate_geojson(feature)
 
     click.echo("✔ valid")
 
 
 @cli.command("add-source")
 @click.argument("username", required=True, type=str)
 @click.argument("id", required=True, type=str)
-@click.argument(
-    "files",
-    required=True,
-    type=click.Path(exists=True, file_okay=True, dir_okay=True),
-    nargs=-1,
-)
+@cligj.features_in_arg
 @click.option("--no-validation", is_flag=True, help="Bypass source file validation")
 @click.option("--token", "-t", required=False, type=str, help="Mapbox access token")
+@click.option("--indent", type=int, default=None, help="Indent for JSON output")
 @click.pass_context
-def add_source(ctx, username, id, files, no_validation, token=None):
+def add_source(ctx, username, id, features, no_validation, token=None, indent=None):
     """Create/add a tileset source
 
     tilesets add-source <username> <id> <path/to/source/data>
     """
     mapbox_api = _get_api()
     mapbox_token = _get_token(token)
-    for f in utils.flatten(files):
-        url = "{0}/tilesets/v1/sources/{1}/{2}?access_token={3}".format(
-            mapbox_api, username, id, mapbox_token
-        )
-        if not no_validation:
-            ctx.invoke(validate_source, source_path=f)
 
-        click.echo(
-            "Adding {0} to mapbox://tileset-source/{1}/{2}".format(f, username, id)
-        )
+    with BytesIO() as io:
+        for feature in features:
+            url = f"{mapbox_api}/tilesets/v1/sources/{username}/{id}?access_token={mapbox_token}"
+            if not no_validation:
+                utils.validate_geojson(feature)
 
-        r = requests.post(url, files={"file": ("tileset-source", open(f, "rb"))})
+            click.echo(
+                f"Adding {feature['geometry']['type']} feature to mapbox://tileset-source/{username}/{id}",
```

Comment:
🔪 can we get rid of this by default? It can be quite verbose with a nominal size geojson. Plus it breaks the pattern of no-formatting-by-default (just print the non-pretty JSON response)